src/test/script_tests.cpp

```diff
@@ -1751,4 +1756,70 @@ BOOST_AUTO_TEST_CASE(script_FindAndDelete)
     BOOST_CHECK(s == expect);
 }
 
+
+BOOST_AUTO_TEST_CASE(script_debugger)
+{
+    CScript testScript = CScript() << 0 << 1;
+    CScript testRedeemScript = CScript() << OP_IF << OP_IF << 1 << OP_ELSE << 2 << OP_ENDIF << OP_ELSE << 3 << OP_ENDIF;
+    BaseSignatureChecker sigChecker;
+    ScriptMachine sm(0, sigChecker);
+
+    bool result = sm.Eval(testScript);
+    BOOST_CHECK(result);
+    sm.BeginStep(testRedeemScript);
+    while (sm.isMoreSteps())
+    {
+        unsigned int pos = sm.getPos();
+        auto info = sm.Peek();
+        if (pos == 4)
+        {
+            BOOST_CHECK(std::get<0>(info) == true);
+            BOOST_CHECK(std::get<1>(info) == OP_2);
+        }
+        // you could print stepping info:
+        // printf("pos %d: %s %s datalen: %d stack len: %d\n", pos, std::get<0>(info) ? "running" : "skipping",
```

Comment:
If memory serves @awemany made the LOG machinery works also while executing a BOOST c++ test. If true I guess we could use LOG("some debug cat", "..") rather than a commented `printf`

Reply:
this comment is more for someone who wants to turn this work into a real script debugger... its suggesting the kind of things that could be printed in every step.